### DISCUSSION TITLE
Accelerate the compilation when working in Parallel

### DISCUSSION THREAD

**Discussion by saidctb** [pyccel/pyccel#1175] [Ideas]
At the moment only the MPI process of rank 0 compiles the file, and if we have a Python module with many independent functions we compile them sequentially,  we would like improve the performance of the compilation by parallelizing it and assigning to each MPI process a list of functions to compile.

**Comment by EmilyBourne** [comment]
Do you have a suggestion for how to achieve this? I think you might struggle to know in advance which functions are independent

**Reply by saidctb** [reply]
Yes we wouldn't calculate it, the user will tell us that the functions are independent , it can be specified by a flag
